src/wooting-rgb-sdk.c

```diff
@@ -213,16 +244,18 @@ bool wooting_rgb_array_set_single(uint8_t row, uint8_t column, uint8_t red, uint
 }
 
 bool wooting_rgb_array_set_full(const uint8_t* colors_buffer) {
-	int baseIndex = 0;
-	for (int row = 0; row < WOOTING_RGB_ROWS; row++) {
-		for (int col = 0; col < WOOTING_RGB_COLS; col++) {
-			uint8_t red = colors_buffer[baseIndex + 0];
-			uint8_t green = colors_buffer[baseIndex + 1];
-			uint8_t blue = colors_buffer[baseIndex + 2];
+	const uint8_t columns = wooting_rgb_cols();
+
+	for (uint8_t row = 0; row < WOOTING_RGB_ROWS; row++) {
+		uint8_t* color = colors_buffer + row * WOOTING_RGB_COLS * 3;
+		for (uint8_t col = 0; col < columns; col++) {
```

Comment:
since i iterate the actual amount of keyboard columns here, i obviously need to change how the rgb values are taken from the buffer.
thus, the current solution